avr/cores/megacommand/CommonTools/helpers.h

```diff
@@ -32,38 +34,39 @@ extern "C" {
 #define _BV32(i) (uint32_t)((uint32_t)(1UL << i))
 
 /** Bit-level access and test macros. **/
-#define SET_BIT(port, bit)   ((port) |= (1 << (bit)))
-#define CLEAR_BIT(port, bit) ((port) &= ~(1 << (bit)))
-#define TOGGLE_BIT(port, bit) ((port) ^= (1 << (bit)))
-#define IS_BIT_SET(port, bit) (((port) & (1 << (bit))) ? 1 : 0)
-#define IS_BIT_CLEAR(port, bit) (((port) & (1 << (bit))) == 0 ? 1 : 0)
+#define SET_BIT(target, bit)                (((uint8_t*)(&target))[(uint8_t)bit / 8U] |=  _bvmasks[(uint8_t)bit & '\07'])
```

Comment:
Presumably the compiler will optimize
```(uint8_t)bit / 8U```
to
```(uint8_t) bit >> 3```

Confirm?

Reply:
Actually no! This is most surprising to me.

Reply:
See the other thread.